IceCertUtils/CertificateUtils.py

```diff
@@ -390,19 +409,34 @@ def createIntermediateFactory(self, alias, *args, **kargs):
         if len(args) > 0 or len(kargs) > 0:
             raise TypeError("unexpected arguments")
 
-        factory = self._createFactory(home = home, dn = dn, altName=altName, parent = self)
+        factory = self._createFactory(home=home,
+                                      dn=dn,
+                                      altName=altName,
+                                      parent=self,
+                                      extendedKeyUsage=extendedKeyUsage,
+                                      crlDistributionPoints=crlDistributionPoints,
+                                      authorityInfoAccess=authorityInfoAccess)
         self.factories[alias] = factory
         return factory
 
-    def getIntermediateFactory(self, alias):
+    def getIntermediateFactory(
```

Comment:
Could `getIntermediateFactory` not create the certificate and be a simple get method instead? Passing factory parameters to get is strange. Do we actually rely on this in any of our scripts that create intermediate certs? If not, I would simplify it.

Reply:
Yes, fixed.